feat: skip manifest generation for tables with unload_strategy=direct-grant (AJDA-1927)

ondrajodas opened the pull request:
# feat: skip manifest generation for tables with unload_strategy=direct-grant (AJDA-1927)

## Summary
Modified the snowflake-transformation component to skip manifest generation and validation for output tables configured with `unload_strategy=direct-grant`. These tables use direct database grants instead of being copied to storage, so they don't need manifest files.

**Changes:**
- Added filtering logic in `SnowflakeTransformationComponent.php` to exclude tables with `unload_strategy=direct-grant` before calling `createManifestMetadata()`
- Applied filtering in **both** the success path (line 47-49) and failure path (line 35-37) to ensure consistent behavior
- Added three comprehensive test cases covering:
  - Direct-grant tables not generating manifests
  - Missing direct-grant tables not causing failures
  - Mixed configurations with both direct-grant and regular tables

### Notes
- The filtering logic is intentionally duplicated in both code paths (success and failure) to ensure consistent behavior
- No changes to ConfigDefinition were needed as the `unload_strategy` field is already supported
- This aligns with how platform-libraries handles direct-grant tables in `TableLoader.php`

---
**Link to Devin run:** https://app.devin.ai/sessions/c6faea752555406394daa0df418c4e78  
**Requested by:** Ondřej Jodas (ondrej.jodas@keboola.com) / @ondrajodas

devin-ai-integration[bot] committed:
feat: skip manifest generation for tables with unload_strategy=direct-grant

- Add filtering logic in both success and failure paths to exclude tables with unload_strategy=direct-grant
- Add test cases to verify direct-grant tables don't generate manifests
- Add test for missing direct-grant tables not causing failures
- Add test for mixed configurations with both direct-grant and regular tables

Implements AJDA-1927

Co-Authored-By: Ondřej Jodas <ondrej.jodas@keboola.com>

ondrajodas commented:
```
# Release Notes
Snowflake Transformation no longer generates manifest files for output tables using `unload_strategy=direct-grant`.

# Plans for Customer Communication
No customer communication is planned.

# Impact Analysis
Only affects configurations with `direct-grant` unload strategy; prevents redundant manifest creation.

# Change Type
Enhancement

# Justification
Ensures consistency with direct-grant logic and avoids unnecessary artifacts.

# Deployment Plan
Standard deployment after merge.

# Rollback Plan
Revert the PR to restore original manifest generation.

# Post-Release Support Plan
None expected.
```

zajca commented on src/SnowflakeTransformationComponent.php:
```diff
@@ -32,7 +32,9 @@ protected function run(): void
-                $config->getExpectedOutputTables(),
+                array_filter($config->getExpectedOutputTables(), function ($table) {
+                    return !isset($table['unload_strategy']) || $table['unload_strategy'] !== 'direct-grant';
+                }),
```
osobně bych teda udělal override té metody getExpectedOutputTables v Configu než to mět tady 2x

ondrajodas replied:
jo, může být

zajca commented on tests/functional/DatadirTest.php:
```diff
@@ -1005,4 +1005,172 @@ public function testEnvVars(): void
+                        [
+                            'source' => 'missing_direct_grant_table',
+                            'destination' => 'out.c-my.missing_direct_grant_table',
+                            'unload_strategy' => 'direct-grant',
```
jak je to se source? Možeš ho vynechat? Protože pro ten direct-grant nedává smysl. Nemá to test proto sa ptám.

ondrajodas replied:
dal jsem ho dopryč aby to bylo jasnější, .... s tím se počítá až při generování manifestu

ondrajodas committed:
Add PHPDoc type annotations for getExpectedOutputTables

Add proper type annotations to satisfy PHPStan requirements and fix
test assertions to work with filtered tables.

ondrajodas committed:
Update DatadirTest to remove source field from direct-grant tables

zajca reviewed (approved):
<img width="600" height="449" alt="Image" src="https://github.com/user-attachments/assets/7b5cddab-80bf-4c61-91b7-a9893c1b6ede" />

ErikZigo reviewed (approved):
lgtm
